### PR TITLE
Reorder and rename test files.

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [x] Have you successfully run `brew tests` with your changes locally?

---

The `--only=` option was changed to account for the new directory-based structure. So e. g. if you call `brew tests --only=utils` it will call `utils_test.rb` **and** all tests under the `utils/` directory.

### CODE REVIEW

**Comment by MikeMcQuaid** [comment]
Filenames seem good, `test/test/` feels weird.

**Comment by jawshooah** [comment]
> Filenames seem good, `test/test/` feels weird.

Agreed. @reitermarkus what's up with that?

**Comment by reitermarkus** [comment]
When globbing only `test/**/*_test.rb`, you also get all of the test files inside `vendor/bundle`. Also, having the test files between all of these is *really* messy:

```
bottles
fixtures
helper
lib
mach
patches
tarballs
Gemfile
Gemfile.lock
testball_bottle.rb
testball.rb
testbottest.rb
testing_env.rb
```

**Comment by jawshooah** [comment]
Seems like `bottles`, `mach`, `patches`, and `tarballs` could all live inside `fixtures`, and `fixtures`, `helper` and `lib` could all live inside `support`. That would clean up the directory quite a bit.

Also not sure why `vendor/bundle` has to live inside `test`. Any reason not to move it up a level?

File: Library/Homebrew/.rubocop_todo.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@ Lint/HandleExceptions:
     - 'extend/pathname.rb'
     - 'formula.rb'
     - 'formula_versions.rb'
-    - 'test/test_ENV.rb'
+    - 'test/test/ENV_test.rb'
```
Comment (MikeMcQuaid) [inline Library/Homebrew/.rubocop_todo.yml]: Again, let's avoid `test/test/*_test.rb`, please.

**Comment by reitermarkus** [comment]
This is ready to merge now.

**Comment by ilovezfs** [comment]
There's a regression with using `test_fixtures("test.png")` in a test block:
https://bot.brew.sh/job/Homebrew%20Core/11589/version=el_capitan/console

Git bisect landed me here.
```
4db1317f38f5a55574854e7bf1dfe4379e807b4f is the first bad commit
commit 4db1317f38f5a55574854e7bf1dfe4379e807b4f
Author: Markus Reiter <me@reitermark.us>
Date:   Mon Oct 24 22:53:25 2016 +0200

    Move all remaining fixtures to `test/support/fixtures`.

:040000 040000 b757a7ded742053020f4af6de583c60b9b8b54af 530d650fc58a1981fd632106bdab1ffa78615e2d M	Library
```

**Comment by ilovezfs** [comment]
Seeing it in https://github.com/Homebrew/homebrew-core/pull/7011 too.

**Comment by MikeMcQuaid** [comment]
@reitermarkus All of Homebrew/brew's and some of Homebrew/homebrew-core's CI has been broken for >24h since this is merged. Can you prioritise fixing this ASAP?

**Comment by ilovezfs** [comment]
It was actually fixed by Josh and Alyssa early I believe

**Comment by MikeMcQuaid** [comment]
From my open PRs builds 45m ago: doesn't seem to be.

**Comment by ilovezfs** [comment]
The fix was in test-bot

**Comment by MikeMcQuaid** [comment]
@ilovezfs Yeh, which was merged 10 hours ago so builds after that time (e.g. 45m ago) should not still be failing.

**Comment by ilovezfs** [comment]
https://github.com/Homebrew/homebrew-core/pull/7008
https://github.com/Homebrew/homebrew-core/pull/7011

were my two PRs that were failing.

After the fixes from Alyssa and Josh were merged, William retriggered CI, and both those PRs passed, so were merged by William and Andrew. That is the extent of my knowledge. What's still failing?

**Comment by MikeMcQuaid** [comment]
Anything on Homebrew/brew: https://bot.brew.sh/job/Brew/2992/version=el_capitan/console

**Comment by MikeMcQuaid** [comment]
Weird, it's sat on an old version of test-bot.

**Comment by ilovezfs** [comment]
Yeah, quite weird, I was not seeing 7f482ba ...

**Comment by ilovezfs** [comment]
I think there's nothing in test-bot actually updating test-bot so anything on 7f482ba is now stranded on 7f482ba.

**Comment by MikeMcQuaid** [comment]
Yeh, fixed it in the CI. All good now @reitermarkus.

**Comment by ilovezfs** [comment]
It looks like there may be another regression stemming from this PR.

In CI for https://github.com/Homebrew/homebrew-science/pull/4619 I'm seeing a failure on Sierra:

https://bot.brew.sh/job/Homebrew%20Science%20Pull%20Requests/6385/version=sierra/console

```
==> brew cleanup --prune=7
==> FAILED
Removing: /Users/brew/Library/Caches/Homebrew/armadillo-7.500.2.sierra.bottle.tar.gz... (4.7M)
Removing: /Users/brew/Library/Caches/Homebrew/arpack-3.4.0.sierra.bottle.1.tar.gz... (471.6K)
Removing: /Users/brew/Library/Caches/Homebrew/autoconf-2.69.sierra.bottle.4.tar.gz... (870.6K)
Removing: /Users/brew/Library/Caches/Homebrew/automake-1.15.sierra.bottle.2.tar.gz... (900.4K)
Removing: /Users/brew/Library/Caches/Homebrew/boost-1.62.0.sierra.bottle.tar.gz... (75.2M)
Removing: /Users/brew/Library/Caches/Homebrew/gcc-6.2.0.sierra.bottle.tar.gz... (82.4M)
Removing: /Users/brew/Library/Caches/Homebrew/gmp-6.1.1.sierra.bottle.tar.gz... (1M)
Removing: /Users/brew/Library/Caches/Homebrew/hdf5-1.8.17.sierra.bottle.1.tar.gz... (3.4M)
Removing: /Users/brew/Library/Caches/Homebrew/isl-0.17.1.sierra.bottle.tar.gz... (1.4M)
Removing: /Users/brew/Library/Caches/Homebrew/libmpc-1.0.3.sierra.bottle.tar.gz... (110.5K)
Removing: /Users/brew/Library/Caches/Homebrew/libtool-2.4.6_1.sierra.bottle.tar.gz... (1008.9K)
Removing: /Users/brew/Library/Caches/Homebrew/libxml2-2.9.4.sierra.bottle.tar.gz... (2.3M)
Removing: /Users/brew/Library/Caches/Homebrew/mpfr-3.1.5.sierra.bottle.tar.gz... (873.6K)
Removing: /Users/brew/Library/Caches/Homebrew/sphinx-doc-1.4.5.sierra.bottle.tar.gz... (9.3M)
Removing: /Users/brew/Library/Caches/Homebrew/szip-2.1.sierra.bottle.1.tar.gz... (43.4K)
Error: uninitialized constant Formulary::FormulaNamespace6c1ff12ed570b061394376cefb27f7cb::Testbottest::TEST_FIXTURE_DIR
Please report this bug:
  https://git.io/brew-troubleshooting
/Users/brew/Library/Caches/Homebrew/Formula/testbottest.rb:4:in `<class:Testbottest>'
/Users/brew/Library/Caches/Homebrew/Formula/testbottest.rb:1:in `load_formula'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:25:in `module_eval'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:25:in `load_formula'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:42:in `load_formula_from_path'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:97:in `load_file'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:88:in `klass'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:84:in `get_formula'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:241:in `factory'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:257:in `from_rack'
/usr/local/Homebrew/Library/Homebrew/cleanup.rb:83:in `block in cleanup_cache'
/usr/local/Homebrew/Library/Homebrew/cleanup.rb:47:in `each'
/usr/local/Homebrew/Library/Homebrew/cleanup.rb:47:in `cleanup_cache'
/usr/local/Homebrew/Library/Homebrew/cleanup.rb:11:in `cleanup'
/usr/local/Homebrew/Library/Homebrew/cmd/cleanup.rb:22:in `cleanup'
/usr/local/Homebrew/Library/Homebrew/brew.rb:94:in `<main>'
```

**Comment by ilovezfs** [comment]
Also for https://github.com/Homebrew/homebrew-science/pull/4620 on all three boxes:
https://bot.brew.sh/job/Homebrew%20Science%20Pull%20Requests/6386/

**Pull request merged**